raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -649,81 +652,172 @@ function receiveTransferExpired(
+      const channel = state.channels[channelKey({ tokenNetwork, partner })];
+      assert(channel?.state === ChannelState.open, 'channel not open');
+
+      assert(confirmation.participant === address, 'participant mismatch');
+      assert(confirmation.chain_id.eq(state.chainId), 'chainId mismatch');
+      assert(confirmation.channel_identifier.eq(channel.id), 'channelId mismatch');
+
+      // don't validate req here, to always update nonce, but do on tx send
+      return withdrawMessage.success({ message: confirmation }, meta);
+    }),
+    catchError((err) => of(withdrawMessage.failure(err, meta))),
+  );
+}
+
+/**
+ * Handles a withdrawExpire.request, sign and send a WithdrawExpired message to partner
+ *
+ * @param state$ - Observable of current state
+ * @param action - WithdrawExpire request which caused this call
+ * @param deps - RaidenEpicDeps members
+ * @param deps.address - Our address
+ * @param deps.log - Logger instance
+ * @param deps.signer - Signer instance
+ * @param deps.config$ - Config observable
+ * @returns Observable of withdrawMessage.request|withdraw.failure actions
+ */
+function sendWithdrawExpired(
+  state$: Observable<RaidenState>,
+  action: withdrawExpire.request,
+  { log, address, signer, config$ }: RaidenEpicDeps,
+): Observable<withdrawExpire.success | withdrawExpire.failure | withdraw.failure> {
+  if (action.meta.direction !== Direction.SENT) return EMPTY;
+  return combineLatest([state$, config$]).pipe(
+    first(),
+    mergeMap(([state, { confirmationBlocks }]) => {
+      const channel = state.channels[channelKey(action.meta)];
+      assert(channel?.state === ChannelState.open, 'channel not open');
+
+      assert(
+        !channel.own.pendingWithdraws.some(
+          (exp) =>
+            exp.type === MessageType.WITHDRAW_EXPIRED &&
+            exp.total_withdraw.eq(action.meta.totalWithdraw) &&
+            exp.expiration.eq(action.meta.expiration),
+        ),
+        'already expired',
+      );
+
+      const req = channel.own.pendingWithdraws.find(
+        (req) =>
+          req.type === MessageType.WITHDRAW_REQUEST &&
+          req.total_withdraw.eq(action.meta.totalWithdraw) &&
+          req.expiration.eq(action.meta.expiration),
+      );
+      assert(req, 'no matching WithdrawRequest found, maybe already confirmed');
+      assert(state.blockNumber > action.meta.expiration + confirmationBlocks, 'not yet expired');
+
+      const expired: WithdrawExpired = {
+        type: MessageType.WITHDRAW_EXPIRED,
+        message_identifier: makeMessageId(),
+        chain_id: req.chain_id,
+        token_network_address: req.token_network_address,
+        channel_identifier: req.channel_identifier,
+        participant: address,
+        total_withdraw: action.meta.totalWithdraw,
+        nonce: channel.own.nextNonce,
+        expiration: req.expiration,
+      };
+      return from(signMessage(signer, expired, { log })).pipe(
+        mergeMap(function* (message) {
```

Comment:
Nope, this would require renaming the var when creating the object literal below; looking at the var type, you'll see `Signed<WithdrawExpired>`, so it's already clear it's the signed message; but I'll keep an eye on opportunities to be more explicit on var naming, thanks